components/modules/crypto.c

```diff
@@ -56,7 +60,38 @@ static int crypto_new_hash_or_hmac(lua_State* L, bool is_hmac) {
     }
 
     if (ainfo == NULL) {
-        return luaL_error(L, "Unsupported algorithm: %s", algo);
+        luaL_error(L, "Unsupported algorithm: %s", algo);
```

Comment:
Style point - all `luaL_error()` calls should be prefixed with `return` to make it clear processing doesn't continue past this point (internally `luaL_error()` raises the exception via a `longjmp()` call, so the stack gets unwound here).

Reply:
This is a now a helper function that returns `const algo_info_t*` so the usual convention can't be applied. It's a pretty small function so there isn't really much chance of confusion I don't think?

Reply:
Ah, but if you do call `luaL_error()` in this function, then *it doesn't return*. That's been a constant source of confusion for people new to integrating with the Lua VM. The best analogy to calling `luaL_error()` is that you're effectively doing a C++ `throw`. Execution of the current function ceases, the stack gets unwound, and the error gets handled somewhere completely different up the call chain.

Not being aware of this mechanic is also an easy source of resource leakage, as a fair bit of due diligence is required to ensure any acquired resources are released prior to calling `luaL_error()`.

Because of this, the canonical way to remind everyone is to prefix all calls to `luaL_error()` with a `return`, even though the return itself never gets actioned.